.github/workflows/lint.yml

```diff
@@ -97,6 +97,12 @@ jobs:
         if: always()
         run: |
           (! git --no-pager grep -In '#!.*python$' -- . || (echo "The above lines have versionless Python shebangs; please specify either python2 or python3"; false))
+      - name: Ensure submodules are up to date
+        env:
+          PR_BODY: ${{ github.event.pull_request.body }}
+          GH_BASE: ${{ github.event.pull_request.base.sha }}
+        if: always()
```

Comment:
Should we actually change this to only run on pull requests? I feel as though this might fail when we land to trunk since the contents of `PR_BODY` and `GH_BASE` would be blank on `push` events

Reply:
```suggestion
        if: ${{ github.event_name == 'pull_request' }}
```

Reply:
makes sense to me. i tested it using `on: pull_request` only, so let me make this change and confirm the test cases still pass.